### PR TITLE
generate gpg keys on the fly

### PR DESCRIPTION
**Pull request by mateusz-bajorek**

### CODE REVIEW

File: .travis/deploy.sh [hunk 1]
```diff
@@ -1,10 +1,40 @@
 #!/usr/bin/env bash
+set -e
+
 if [ ! -z "$TRAVIS_TAG" ]
 then
     echo "on a tag -> set pom.xml <version> to $TRAVIS_TAG"
-    mvn --settings .travis/mvnsettings.xml org.codehaus.mojo:versions-maven-plugin:2.3:set -DnewVersion=$TRAVIS_TAG 1>/dev/null 2>/dev/null
+    mvn --settings .travis/mvnsettings.xml org.codehaus.mojo:versions-maven-plugin:2.3:set -DnewVersion=$TRAVIS_TAG
```
Comment (JasonGermaine) [inline .travis/deploy.sh:7]: should this not be the only command within the `if` as the other commands seems common between both branches from what I can see

Comment (mateusz-bajorek) [reply .travis/deploy.sh:7]: True, it should :D

File: .travis/gpg.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# cleanup local configuration
+shred gen-key-script
+
+# publish the gpg key
+# (use keyserver.ubuntu.com as travis request keys from this server,
+#  we avoid synchronization issues, while releasing)
+gpg --keyserver keyserver.ubuntu.com --send-keys ${GPG_KEYNAME}
+
+# wait for the key beeing accessible
+while(true); do
```
Comment (JasonGermaine) [inline .travis/gpg.sh:45]: should we wrap within a `timeout`?

Comment (mateusz-bajorek) [reply .travis/gpg.sh:45]: Travis build will timeout itself if nothing happens in the build.

**Review by JasonGermaine — APPROVED** [review]
👍

**Review by garethpelly — APPROVED** [review]

**Comment by garethpelly** [comment]
:+1:

**Pull request merged**